### DISCUSSION TITLE
Is a good idea to develop locally with with kubernetes ?

### DISCUSSION THREAD

**Discussion by ruslan-byondxr** [nrwl/nx#7399] [General]
NX is great!  our nx monorepo consists of react apps and libs, graphql-express microservices, MongoDB managers, configurations, shared typescript interfaces, and schemas.  
We have some architectural discussions in our team. One suggestion is that we should develop our microservices apps with Kubernetes and scaffold,  which is a command-line tool that facilitates continuous development for Kubernetes applications.
 https://github.com/GoogleContainerTools/skaffold
 
What do you think guys, It's a good idea ? ... Does anyone have experience using NX local development with dockers and Kubernetes?

**Comment by michaelparkadze** [comment]
Looking for the same answer, currently we are using skaffold and basically developing locally for kubernetes and I am trying to understand how does nx/turborepo comes into play.

**Comment by rfprod** [comment]
> Does anyone have experience using NX local development with dockers and Kubernetes?

I would recommend considering https://github.com/werf/werf among possible options.
There is an article on local development using werf https://blog.flant.com/local-development-in-kubernetes-with-werf/
